aioshelly/coap.py

```diff
@@ -69,6 +86,74 @@ def socket_init():
     return sock
 
 
+def verify_l2socket_creation_permission():
+    """Create a socket using the scapy configured l2socket.
+
+    Try to create the socket
+    to see if we have permissions
+    since sniff will do it another
+    thread so we will not be able to capture
+    any permission or bind errors.
+    """
+    conf.L2socket()
```

Comment:
Do we have to close down this socket when you've checked permissions?

Reply:
Done.
You should review also https://github.com/home-assistant/core/blob/0382c932836ff58854fb2076a16b825928f95566/homeassistant/components/dhcp/__init__.py#L276 ;-)